build-tools/mono-runtimes/mono-runtimes.targets

```diff
@@ -296,6 +304,51 @@
     <Touch
         Files="@(_RuntimeSource);@(_ProfilerSource);@(_MonoPosixHelperSource);@(_BclProfileItems);@(_MonoBtlsSource)"
     />
+    <Exec
+        Command="make $(MakeConcurrency) -i do-test PROFILE=monodroid"
```

Comment:
I'd use `test` instead of `do-test`, it's the more canonical target.

Reply:
also do you really need -i? which errors do you get?

Reply:
@akoeplinger: [Yes, I really need `-i`](https://xamarinhq.slack.com/archives/C03CELHQQ/p1505411469000076):

```
^[[Dmake[6]: *** No rule to make target `Test/../../../../external/api-doc-tools/monodoc/Test/Monodoc.Ecma/EcmaUrlTests.cs', needed by `net_4_x_monodoc_test.dll'.  Stop.
```

Alexander Kyte suggested `make do-trst`, and Zoltan mentioned that Mono Jenkins uses `-i`.

Reply:
Ah, I see now you're compiling from mcs/class so you compile all assemblies even those that don't apply to you (like monodoc in this case, though still weird that it can't find the file in the submodule).

Please switch over to running `make -i test` in `$(MonoSourceFullPath)\runtime`, this will only compile the tests of assemblies which are included in the monodroid profile.

Ignoring errors shouldn't be required but it is right now, because the following test assemblies don't build in your profile (something to look at):

```
make[3]: [monodroid_System.Security_test.dll] Error 1 (ignored)
make[3]: [monodroid_System.IdentityModel_test.dll] Error 1 (ignored)
make[3]: [monodroid_System.ServiceModel_test.dll] Error 1 (ignored)
make[3]: [monodroid_System.Web.Services_test.dll] Error 1 (ignored)
make[3]: [monodroid_System.ComponentModel.DataAnnotations_test.dll] Error 1 (ignored)
make[3]: [monodroid_Mono.Posix_test.dll] Error 1 (ignored)
```